bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/discovery/HiveDiscoveryService.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.discovery;
+
+import static org.openhab.binding.hive.internal.HiveBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.config.discovery.ScanListener;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.openhab.binding.hive.internal.dto.HiveNode;
+import org.openhab.binding.hive.internal.handler.HiveBridgeHandler;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Chris Foot - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.hive")
+@NonNullByDefault
+public class HiveDiscoveryService extends AbstractDiscoveryService implements DiscoveryService, ThingHandlerService {
+
+    @Nullable
```

Comment:
The Nullable annotations should on the same line as the field. e.g.: ` protected @Nullable ...`

Reply:
I wrote them like that but for some reason eclipse moves them every time I save!

Reply:
Found the setting in the openHab formatter (New Lines, on Fields). It looks like it may be wrong by default